spec.md

```diff
@@ -936,6 +937,22 @@ module Top:
   node consumer_debug = read(c.out.cref); ; Consumer-side signal
 ```
 
+## Property Types
+
+FIRRTL property types represent information about the circuit that is not
+hardware. This is useful to capture domain specific knowledge and design intent
+alongside the hardware description within the same FIRRTL.
+
+Property types cannot affect hardware functionality or the hardware ABI. They
+cannot be used in any hardware types, including aggregates. They only exist to
+augment the hardware description with extra information.
+
+Handling of property types is completely implementation defined. A valid FIRRTL
+compiler implementation may do anything with property types as long as the
+existence of property types does not affect hardware functionality or the
```

Comment:
Looks like this was intentionally phrased to allow properties to change the hardware emitted, just not what's observable externally re:ABI and functionality? Sounds good/right to me.